[0.76] `NU1004: The package references have changed for native` with New Arch enabled

### Problem Description

Building RNW with New Architecture enabled, I get this error:

` × Build failed with message 1>D:\react-native-test-app\node_modules\react-native-windows\Microsoft.ReactNative\Microsoft.ReactNative.vcxproj : error NU1004: The package references have changed for native. Lock file's package references: boost:[1.83.0, ), Microsoft.JavaScript.Hermes:[0.1.23, ), Microsoft.SourceLink.GitHub:[1.1.1, ), Microsoft.Windows.CppWinRT:[2.0.230706.1, ), Microsoft.WindowsAppSDK:[1.5.240227000, ), project's package references: boost:[1.83.0, ), Microsoft.JavaScript.Hermes:[0.1.23, ), Microsoft.Windows.CppWinRT:[2.0.230706.1, ), Microsoft.WindowsAppSDK:[1.5.240227000, ).The packages lock file is inconsistent with the project dependencies so restore can't be run in locked mode. Disable the RestoreLockedMode MSBuild property or pass an explicit --force-evaluate option to run restore to update the lock file. [D:\react-native-test-app\example\windows\Example.sln]. Check your build configuration.`

Where `Microsoft.SourceLink.GitHub` is no longer present. Removing `node_modules/react-native-windows/Microsoft.ReactNative/packages.fabric.lock.json` gets me further.

Looking in the npm package, I see `packages.fabric.lock.json` and `packages.lock.json` are present. Should they be included in the package?

### Steps To Reproduce

n/a

### Expected Results

_No response_

### CLI version

14.1.1

### Environment

```markdown
n/a
```


### Community Modules

_No response_

### Target Platform Version

10.0.22621

### Target Device(s)

Desktop

### Visual Studio Version

Visual Studio 2022

### Build Configuration

Debug

### Snack, code example, screenshot, or link to a repository

_No response_

--- a/change/@react-native-windows-automation-channel-618f52b6-bbe6-4449-9165-d6a447f0da53.json
+++ b/change/@react-native-windows-automation-channel-618f52b6-bbe6-4449-9165-d6a447f0da53.json
@@ -1,0 +1,7 @@
+{
+  "type": "patch",
+  "comment": "Exclude NuGet package lock files from NPM packages",
+  "packageName": "@react-native-windows/automation-channel",
+  "email": "jthysell@microsoft.com",
+  "dependentChangeType": "patch"
+}

--- a/change/react-native-windows-73248300-7bfa-4031-8b40-14125d248e4f.json
+++ b/change/react-native-windows-73248300-7bfa-4031-8b40-14125d248e4f.json
@@ -1,0 +1,7 @@
+{
+  "type": "prerelease",
+  "comment": "Exclude NuGet package lock files from NPM packages",
+  "packageName": "react-native-windows",
+  "email": "jthysell@microsoft.com",
+  "dependentChangeType": "patch"
+}

--- a/packages/@react-native-windows/automation-channel/package.json
+++ b/packages/@react-native-windows/automation-channel/package.json
@@ -37,7 +37,9 @@
   },
   "files": [
     "lib-commonjs",
-    "windows"
+    "windows",
+    "!packages.lock.json",
+    "!packages.fabric.lock.json"
   ],
   "engines": {
     "node": ">= 18"

--- a/vnext/package.json
+++ b/vnext/package.json
@@ -144,7 +144,9 @@
     "/metro-react-native-platform.js",
     "/metro.config.js",
     "/react-native.config.js",
-    "/rn-get-polyfills.js"
+    "/rn-get-polyfills.js",
+    "!packages.lock.json",
+    "!packages.fabric.lock.json"
   ],
   "promoteRelease": true,
   "engines": {

